### PR TITLE
Wait for profile refresh to complete in `.refreshable {}`

### PR DESCRIPTION
**Pull request by bfollington**
# Changes

- Created `.refresh()` helper that is used by the model and the UI

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift [hunk 1]
```diff
@@ -162,6 +163,47 @@ struct EditProfileSheetCursor: CursorProtocol {
     }
 }
 
+extension Store<UserProfileDetailModel> {
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift:166]: Clever

File: xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift [hunk 2]
```diff
@@ -162,6 +163,47 @@ struct EditProfileSheetCursor: CursorProtocol {
...
+    func refresh() async {
+        guard let address = state.user?.address else {
+            return
+        }
+        
+        // Ensure updates are sent on the main thread
+        let send = { action in
+            DispatchQueue.main.async {
+                self.send(action)
+            }
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift:176]: The new way:

```
Task { @MainActor in self.send(action) }
```

**Review by gordonbrander — APPROVED** [review]
One non-blocking suggestion.

File: xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift (hunk 2 above)
Comment (bfollington) [reply xcode/Subconscious/Shared/Components/Detail/UserProfileDetailView.swift:176]: Urgh I was looking for something like this.

**Pull request merged**